kong/cmd/utils/prefix_handler.lua

```diff
@@ -247,8 +247,8 @@ local function prepare_prefix(kong_config, nginx_custom_template_path)
   end
 
   -- generate default SSL certs if needed
-  if kong_config.proxy_ssl_enabled and not kong_config.ssl_cert and
-     not kong_config.ssl_cert_key then
+  if (#kong_config.stream_listeners > 0 or kong_config.proxy_ssl_enabled) and
```

Comment:
`kong_config.stream_listeners[1] ~= nil` is probably a better check.

Reply:
Changed.